Source/santad/SNTSyncdQueue.mm

```diff
@@ -74,34 +74,44 @@ - (void)reassessSyncServiceConnection {
       STRONGIFY(self);
       SNTConfigurator *configurator = [SNTConfigurator configurator];
 
+      NSURL *newSyncBaseURL = [configurator syncBaseURL];
+      BOOL statsCollectionEnabled = [configurator enableStatsCollection];
+
+      // Don't allow an empty string for the sync base URL
+      if (newSyncBaseURL && newSyncBaseURL.absoluteString.length == 0) {
```

Comment:
```suggestion
      if (newSyncBaseURL) {
```

The 2nd condition is effectively impossible

Reply:
It is as long as someone doesn't change the configurator in the future... If you're arguing to remove the length check, we'd just remove the whole if statement. I think I'd rather keep it in defensively, but agree it shouldn't be possible in the current codebase.

Reply:
We typically do `.length` checks on strings because it catches both nil and 0-length on NSString, but because syncBaseURL is already checking for both of those conditions the extra check is unnecessary. If we're worried about someone changing configurator, we should add a comment in configurator to explain why the checks are there.